### DISCUSSION TITLE
Query string processing

### DISCUSSION THREAD

**Discussion by LionelWalker** [cjvandyk/Quix#4] [Ideas]
Does extensions handle query strings? It be slick if I could parse a query string into a dictionary for later manipulation.

**Comment by cjvandyk** [comment]
I took a quick look and the System.Web.HttpUtility has a ParseQueryString() method that returns a NameValueCollection which could be used.
https://docs.microsoft.com/en-us/dotnet/api/system.web.httputility.parsequerystring?view=net-5.0
The Microsoft.AspNetCore.WebUtilities.QueryHelpers has a ParseQuery() method that returns a Dictionary<string, string> which could also be used if you were in ASP.NET.
https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.webutilities.queryhelpers.parsequery?view=aspnetcore-5.0

**Comment by LionelWalker** [comment]
How bout other direction making query string from dictionary?

**Reply by cjvandyk** [reply]
If you pull the latest source, you'll find that I've added QueryStringToDictionary().  It'll be in the next NuGet release, but if you wanted it early, you can pull the code and build. ;-)

https://github.com/cjvandyk/Quix/blob/b0dea1bc983ab1b12ee7801aa3be9357f1135f53/Extensions/Classes/System.String.cs#L1070

**Reply by cjvandyk** [reply]
> How bout other direction making query string from dictionary?

Thank you for the suggestion.  As of v2.1.480 (published on 2020/02/02) this feature was added.
